src/Io/Transaction.php

```diff
@@ -66,10 +79,14 @@ protected function next(RequestInterface $request)
             $promise = $promise->then(array($that, 'bufferResponse'));
         }
 
-        return $promise->then(
-            function (ResponseInterface $response) use ($request, $that) {
-                return $that->onResponse($response, $request);
-            }
+        return Promise\Timer\timeout(
```

Comment:
This only times out a single request within this "transaction". In other words: This timeout will be restarted for each redirection and the consumer has no control over applying a total timeout.

It's my understanding that this timeout should apply to the whole "transaction", i.e. simply move this up to the `send()` method instead of the `next()` method.

What do you think about this?